### DISCUSSION TITLE
Issue while deploying ERC 20 contract

### DISCUSSION THREAD

**Discussion by 0xtdey** [Cyfrin/foundry-full-course-cu#271] [Q&A] [answered]
<img width="689" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/e1ed0195-2743-4378-ac42-0870f1b1929a">

I have all the directories set accordingly but still cannot figure out why it shows error in directory.

**Comment by shikhar229169** [comment]
Ensure when you import ERC20 contract, you wrote this:
import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";

According to the error you only wrote:
import {ERC20} from "@openzeppelin/contracts/token/ERC20";

**Reply by 0xtdey** [reply]
I wrote this in the OurToken.sol contract 
<img width="605" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/9c8041d2-e817-47ee-a4e7-8a315ea49baa">

**Reply by 0xtdey** [reply]
And this is my foundry.toml file 

<img width="653" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/f62b17e7-a824-47c3-8690-d2e097ba4802">

**Reply by shikhar229169** [reply]
After ERC20 write /ERC20.sol

**Reply by shikhar229169** [reply]
Or what you can do is write import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";

and in foundry.toml
```
remappings = ["@openzeppelin=lib/openzeppelin-contracts"]
```

**Reply by 0xtdey** [reply]
I did the following changed but then it shows the red squiggly in the code 
<img width="573" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/a5c463d0-1b6a-469e-a3f1-1eda17de1d62">
and it shows this as the error
<img width="671" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/6809e40c-d1a6-4387-bc08-ce1958c2b509">
Same thing it cannot find the directory properly

**Reply by shikhar229169** [reply]
What is your remappings please show that.

Copy paste this one: 
```remappings = ["@openzeppelin=lib/openzeppelin-contracts"]```

**Reply by Naman1729** [reply]
hey @tamoghna-dey , I think you haven't install **openzeppelin-contracts** . so run this command :

```bash
forge install OpenZeppelin/openzeppelin-contracts --no-commit
```

then , check this folder "**openzeppelin-contracts**" is now present in lib folder.

**Reply by 0xtdey** [reply]
yaa I already have it installed bro 
<img width="219" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/62704bce-ec5f-4267-a301-aef6d62bd94e">

**Reply by 0xtdey** [reply]
The thing is if I import like this it doesn't show error in the code 
<img width="586" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/6a0cac8b-6b5b-4ab3-946b-4ba820da0a73">
But when i deploy it, it shows it cannot fetch the directory properly

**Reply by shikhar229169** [reply]
But you have to enter /ERC20.sol in the end.
Because you have to mention from which solidity file you need that ERC20 contract.
But you are only mentioning it till the ERC20 folder.

**Reply by Naman1729** [reply]
After installation follow @shikhar229169 instructions  
for import :
```bash
import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
In foundry.toml :
```bash
remappings = ["@openzeppelin=lib/openzeppelin-contracts"]
```

**Reply by ChefAharoni** [reply, marked as answer]
Remappings can be tricky, foundry is a little bit buggy in VSCode.

Try these [solutions](https://github.com/Cyfrin/foundry-full-course-f23/discussions/194) regarding mappings issues.

For me, writing the remappings in a .txt file helped, this is how it looks in mine:
<img width="378" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/4399057/dda8f167-0918-4206-8ea1-538e980e49ed">

```text
@openzeppelin/=lib/openzeppelin-contracts/
```

You can also run `forge remappings` to see the remappings foundry is recognizing.

**Reply by 0xtdey** [reply]
Thanks, @ChefAharoni and @shikhar229169 combining both of you guys suggestions it worked out.  👍
